### PR TITLE
O3-1106: Remove 'isTablet' workspace prop

### PR DESCRIPTION
**Pull request by Piumal1999**
## Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [x] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [x] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary

Removed `tablet` workspace prop.
Obtained `isTablet` with `const isTablet = useLayoutType() === 'tablet';`

Note: I also did the same for `AppointmentsForm`, `VisitTypeOverview`, `VitalsAndBiometricForm` components which don't use the `defaultWorkspace`.

## Screenshots

*None.*

## Issue

https://issues.openmrs.org/browse/O3-1106

## Other

*None.*

### CODE REVIEW

**Comment by Piumal1999** [comment]
I completed the issue. But the builds are failing after caaf300 commit.

File: packages/esm-patient-appointments-app/src/appointments/appointments-form.component.tsx [hunk 1]
```diff
@@ -14,21 +14,21 @@ import {
...
 import { amPm, convertTime12to24 } from '@openmrs/esm-patient-common-lib';
-import { useLocations, useSessionUser, showToast, showNotification } from '@openmrs/esm-framework';
+import { useLocations, useSessionUser, showToast, showNotification, useLayoutType } from '@openmrs/esm-framework';
 import { appointmentsSearchUrl, createAppointment, useAppointmentService } from './appointments.resource';
 import styles from './appointments-form.scss';
 import { AppointmentPayload } from '../types';
 
 interface AppointmentsFormProps {
   closeWorkspace: () => {};
-  isTablet: boolean;
```
Comment (brandones) [inline packages/esm-patient-appointments-app/src/appointments/appointments-form.component.tsx:24]: Actually, would you mind deleting these custom interfaces and using `DefaultWorkspaceProps` when applicable?

File: packages/esm-patient-chart-app/src/visit/visit-form/visit-form.component.tsx [hunk 2]
```diff
@@ -32,13 +32,15 @@ import {
...
+  useLayoutType,
 } from '@openmrs/esm-framework';
 import { amPm, convertTime12to24, DefaultWorkspaceProps } from '@openmrs/esm-patient-common-lib';
 import VisitTypeOverview from './visit-type-overview.component';
 import styles from './visit-form.component.scss';
 
-const StartVisitForm: React.FC<DefaultWorkspaceProps> = ({ isTablet, patientUuid, closeWorkspace }) => {
+const StartVisitForm: React.FC<DefaultWorkspaceProps> = ({ patientUuid, closeWorkspace }) => {
   const { t } = useTranslation();
+  const isTablet = useLayoutType() === 'tablet';
```
Comment (brandones) [inline packages/esm-patient-chart-app/src/visit/visit-form/visit-form.component.tsx:43]: Looks like this isn't needed here

**Review by brandones — CHANGES_REQUESTED** [review]
Awesome work. Two requests, plus one thing: you also need to remove the place where `isTablet` gets passed to workspaces. I think this is in `workspace.ts`.

**Comment by brandones** [comment]
Build should be fixed since https://github.com/openmrs/openmrs-esm-patient-management/pull/111

File: packages/esm-patient-chart-app/src/visit/visit-form/visit-form.component.tsx (hunk 2 above)
Comment (Piumal1999) [reply packages/esm-patient-chart-app/src/visit/visit-form/visit-form.component.tsx:43]: It is used in line 230. (`<ButtonSet className={isTablet ? styles.tablet : styles.desktop}>`)

File: packages/esm-patient-appointments-app/src/appointments/appointments-form.component.tsx (hunk 1 above)
Comment (Piumal1999) [reply packages/esm-patient-appointments-app/src/appointments/appointments-form.component.tsx:24]: Ah right. I'll do it and update the PR

File: packages/esm-patient-common-lib/src/types/workspace.ts [hunk 3]
```diff
@@ -11,5 +11,4 @@ export enum WorkspaceWindowState {
 export interface DefaultWorkspaceProps {
   closeWorkspace(): void;
   patientUuid: string;
-  isTablet: boolean;
```
Comment (Piumal1999) [inline packages/esm-patient-common-lib/src/types/workspace.ts:14]: > you also need to remove the place where isTablet gets passed to workspaces. I think this is in workspace.ts.

@brandones is this the one you mentioned?

File: packages/esm-patient-appointments-app/src/appointments/appointments-form.component.tsx (hunk 1 above)
Comment (Piumal1999) [reply packages/esm-patient-appointments-app/src/appointments/appointments-form.component.tsx:24]: Updated with f3582be

Comment (Piumal1999) [reply packages/esm-patient-appointments-app/src/appointments/appointments-form.component.tsx:24]: Updated with f3582be

File: packages/esm-patient-common-lib/src/types/workspace.ts (hunk 3 above)
Comment (brandones) [reply packages/esm-patient-common-lib/src/types/workspace.ts:14]: No, I mean here: https://github.com/openmrs/openmrs-esm-patient-chart/blob/4a7fb821c830a0cd055d8b453f9186177586420d/packages/esm-patient-chart-app/src/workspace/workspace-renderer.component.tsx#L40 .

Comment (Piumal1999) [reply packages/esm-patient-common-lib/src/types/workspace.ts:14]: It is also already updated

Comment (brandones) [reply packages/esm-patient-common-lib/src/types/workspace.ts:14]: You're right! Sorry about that.

**Comment by brandones** [comment]
Build should be fixed by https://github.com/openmrs/openmrs-esm-patient-chart/pull/561

**Comment by brandones** [comment]
@Piumal1999 This looks great, can you merge master and fix the conflicts? Merging master should hopefully also fix the build error.

**Review by brandones — APPROVED** [review]

**Comment by brandones** [comment]
Thank you @Piumal1999 , this is really helpful!

**Pull request merged**